src/tests/suite/daos_obj_array.c

```diff
@@ -579,9 +579,9 @@ array_dkey_punch_enumerate(void **state)
 	print_message("DONE DKEY Enumeration (%d extents) -------\n", total_nr);
 	assert_int_equal(total_nr, 100);
 
-	/** punch first 10 records */
+	/** punch last 10 dkeys, and another 10 non-existent dkeys */
 	print_message("Punching 10 dkeys...\n");
-	for (i = 0; i < 10; i++) {
+	for (i = 90; i < 110; i++) {
```

Comment:
Not necessarily in this patch but it would be better to use macros

for (i = (NUM_KEYS - KEYS2PUNCH); i < (NUM_KEYS + NONEXISTENT_KEYS); i++) {
    ....
}

Then it's self documenting to some extent and easy to change without modifying 10 places